File: Assets/src/model/items/content/ItemWildwoodLeaf.cs
using System;
using System.Linq;

[Serializable]
class ItemWildwoodLeaf : Item, IEdible {
  public static int yieldCost = 2;
  public override int stacksMax => int.MaxValue;

  public void Eat(Actor a) {
    a.statuses.Add(new StatusWild());
    stacks--;
  }

  internal override string GetStats() => "Apply the Wild status for 15 turns, doubling your movespeed.";
}

[System.Serializable]
internal class StatusWild : StackingStatus, IActionCostModifier {
  public StatusWild(int stacks) : base(stacks) {}
  public StatusWild() : this(15) {}

  public override string Info() => $"Move twice as fast.";

  public ActionCosts Modify(ActionCosts input) {
    input[ActionType.MOVE] /= 2;
    return input;
  }

  public override void Step() {
    stacks--;
  }
}

File: web/src/model/items/ItemWildwoodLeaf.ts
import { Item, STACKABLE_TAG, EDIBLE_TAG, type IStackable, type IEdible } from '../Item';
import type { Actor } from '../Actor';
import { StatusWild } from '../statuses/StatusWild';

/**
 * Stackable edible. Eating grants StatusWild (2x move speed for 15 turns).
 * Port of C# ItemWildwoodLeaf.cs.
 */
export class ItemWildwoodLeaf extends Item implements IStackable, IEdible {
  readonly [STACKABLE_TAG] = true as const;
  readonly [EDIBLE_TAG] = true as const;

  readonly stacksMax = 10;
  private _stacks: number;

  get stacks(): number {
    return this._stacks;
  }

  set stacks(value: number) {
    if (value < 0) throw new Error('Setting negative stack! ' + this + ' to ' + value);
    this._stacks = value;
    if (this._stacks === 0) {
      this.Destroy();
    }
  }

  constructor(stacks = 1) {
    super();
    this._stacks = stacks;
  }

  eat(actor: Actor): void {
    actor.statuses.add(new StatusWild());
    this.stacks--;
  }

  getStats(): string {
    return 'Apply the Wild status for 15 turns, doubling your movespeed.';
  }
}
